File: server/bench/fixtures/sample.go
package sample

import "fmt"

type User struct {
	Name string
	Age  int
}

type Repository struct {
	users []User
}

func NewRepository(users []User) *Repository {
	return &Repository{users: users}
}

func (r *Repository) Find(name string) (*User, bool) {
	for i := range r.users {
		if r.users[i].Name == name {
			return &r.users[i], true
		}
	}
	return nil, false
}

func (r *Repository) Count() int {
	return len(r.users)
}

func Greet(u User) string {
	return fmt.Sprintf("Hello, %s!", u.Name)
}

func Run() {
	repo := NewRepository([]User{{Name: "alice", Age: 30}, {Name: "bob", Age: 25}})
	if u, ok := repo.Find("alice"); ok {
		fmt.Println(Greet(*u))
	}
	fmt.Printf("total: %d\n", repo.Count())
}

File: server/bench/fixtures/sample.rb
module Sample
  class User
    attr_reader :name, :age

    def initialize(name, age)
      @name = name
      @age = age
    end
  end

  class Repository
    def initialize(users)
      @users = users
    end

    def find(name)
      @users.find { |u| u.name == name }
    end

    def count
      @users.length
    end
  end

  def self.greet(user)
    "Hello, #{user.name}!"
  end
end

repo = Sample::Repository.new([
  Sample::User.new("alice", 30),
  Sample::User.new("bob", 25),
])

found = repo.find("alice")
puts Sample.greet(found) if found
puts "total: #{repo.count}"
